nni/algorithms/nas/pytorch/fbnet/utils.py

```diff
@@ -0,0 +1,509 @@
+        return layer_configs, layer_in_shapes
+
+    def _create_perfs(self, cnt_of_runs=200):
+        """Create performance cost for each op."""
+        if self.perf_metric == "latency":
+            self.lut_perf = self._calculate_latency(cnt_of_runs)
+        elif self.perf_metric == "flops":
+            self.lut_perf = self._calculate_flops()
+
+        self._write_lut_to_file()
+
+    def _calculate_flops(self, eps=0.001):
+        """FLOPs cost."""
+        flops_lut = [{} for i in range(self.cnt_layers)]
+        layer_id = 0
+
+        for stage_name in self.lut_ops:
+            stage_ops = self.lut_ops[stage_name]
+            ops_num = self.layer_num[stage_name]
+
+            for _ in range(ops_num):
+                for op_name in stage_ops:
+                    layer_config = self.layer_configs[layer_id]
+                    key_params = {"fm_size": layer_config[3]}
+                    op = stage_ops[op_name](*layer_config[0:3], **key_params)
+
+                    # measured in micro-second
+                    flops = count_flops(op, self.layer_in_shapes[layer_id])
+                    flops = eps if flops == 0.0 else flops
+                    flops_lut[layer_id][op_name] = float(flops)
+                layer_id += 1
+
+        return flops_lut
+
+    def _calculate_latency(self, cnt_of_runs):
+        """Latency cost."""
+        LATENCY_BATCH_SIZE = 1
+        latency_lut = [{} for i in range(self.cnt_layers)]
+        layer_id = 0
+
+        for stage_name in self.lut_ops:
+            stage_ops = self.lut_ops[stage_name]
+            ops_num = self.layer_num[stage_name]
+
+            for _ in range(ops_num):
+                for op_name in stage_ops:
+                    layer_config = self.layer_configs[layer_id]
+                    key_params = {"fm_size": layer_config[3]}
+                    op = stage_ops[op_name](*layer_config[0:3], **key_params)
+                    input_data = torch.randn(
+                        (LATENCY_BATCH_SIZE, *self.layer_in_shapes[layer_id])
+                    )
+                    globals()["op"], globals()["input_data"] = op, input_data
+                    total_time = timeit.timeit(
+                        "output = op(input_data)",
+                        setup="gc.enable()",
+                        globals=globals(),
+                        number=cnt_of_runs,
+                    )
+                    # measured in micro-second
+                    latency_lut[layer_id][op_name] = (
+                        total_time / cnt_of_runs / LATENCY_BATCH_SIZE * 1e6
+                    )
+                layer_id += 1
+
+        return latency_lut
+
+    def _write_lut_to_file(self):
+        """Save lut as numpy file."""
+        np.save(self.lut_file, self.lut_perf)
+
+    def _load_from_file(self):
+        """Load numpy file."""
+        self.lut_perf = np.load(self.lut_file, allow_pickle=True)
+
+
+def count_flops(model=None, in_shape=(3, 112, 112), multiply_adds=False):
```

Comment:
Is it possible to leverage `flops_counter` provided by NNI?

Reply:
The count_flops return the flops with only multiplies, may be OK.

Reply:
The count_flops return the flops with only multiplies, may be OK with flops_counter.

Reply:
Hi, I have not find the flops_counter definition in NNI. I have seen that the lib 'ptflops' is used in Cream example, the pickle file with flops is used in SOPS. Thanks:)

Reply:
It's here: https://github.com/microsoft/nni/blob/master/nni/compression/pytorch/utils/counter.py

If there are subtle differences that matter, please go ahead and keep yours.

Reply:
Done, the function is replaced with 'count_flops_params' in NNI.